addon/src/setup-rendering-context.ts

```diff
@@ -1,27 +1,33 @@
-/* globals EmberENV */
 import { run } from '@ember/runloop';
 import Ember from 'ember';
-import global from './global';
+import global from './global.ts';
 import {
   type BaseContext,
   type TestContext,
   isTestContext,
   getContext,
-} from './setup-context';
-import settled from './settled';
-import { hbs } from 'ember-cli-htmlbars';
-import getRootElement from './dom/get-root-element';
-import type { Owner } from './build-owner';
-import getTestMetadata from './test-metadata';
+} from './setup-context.ts';
+import settled from './settled.ts';
+import getRootElement from './dom/get-root-element.ts';
+import type { Owner } from './build-owner.ts';
+import getTestMetadata from './test-metadata.ts';
 import { assert } from '@ember/debug';
-import { runHooks } from './helper-hooks';
-import hasEmberVersion from './has-ember-version';
-import isComponent from './-internal/is-component';
-import { ComponentRenderMap, SetUsage } from './setup-context';
-
-const OUTLET_TEMPLATE = hbs`{{outlet}}`;
-const EMPTY_TEMPLATE = hbs``;
-const INVOKE_PROVIDED_COMPONENT = hbs`<this.ProvidedComponent />`;
+import { runHooks } from './helper-hooks.ts';
+import hasEmberVersion from './has-ember-version.ts';
+import isComponent from './-internal/is-component.ts';
+import { ComponentRenderMap, SetUsage } from './setup-context.ts';
+
+// the built in types do not provide types for @ember/template-compilation
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { precompileTemplate } from '@ember/template-compilation';
+
+const OUTLET_TEMPLATE = precompileTemplate(`{{outlet}}`, { strictMode: false });
```

Comment:
I didn't want to import from ember-cli-htmlbars, so I used `@ember/template-compilation` for `precompileTemplate`

Reply:
This could also be pulled out to a separate PR, if folks want